File: app/src/test/java/com/mohsenoid/rickandmorty/test/LocationDataFactory.java
package com.mohsenoid.rickandmorty.test;

import com.mohsenoid.rickandmorty.data.db.dto.DbLocationModel;
import com.mohsenoid.rickandmorty.data.network.dto.NetworkLocationModel;
import com.mohsenoid.rickandmorty.domain.entity.LocationEntity;

import org.jetbrains.annotations.NotNull;

public class LocationDataFactory {

    private LocationDataFactory() { /* this will prevent making a new object of this type */ }

    public static class Db {

        @NotNull
        public static DbLocationModel makeDbLocationModel() {
            return new DbLocationModel(DataFactory.randomString(), DataFactory.randomString());
        }
    }

    public static class Network {

        @NotNull
        public static NetworkLocationModel makeNetworkLocationModel() {
            return new NetworkLocationModel(DataFactory.randomString(), DataFactory.randomString());
        }
    }

    public static class Entity {

        @NotNull
        public static LocationEntity makeLocationEntity() {
            return new LocationEntity(DataFactory.randomString(), DataFactory.randomString());
        }
    }
}

File: app/src/test/kotlin/com/mohsenoid/rickandmorty/test/LocationDataFactory.kt
package com.mohsenoid.rickandmorty.test

import com.mohsenoid.rickandmorty.data.api.model.ApiLocation
import com.mohsenoid.rickandmorty.data.db.model.DbLocation
import com.mohsenoid.rickandmorty.domain.model.ModelLocation

object LocationDataFactory {

    internal fun makeDbLocation(): DbLocation =
        DbLocation(
            name = DataFactory.randomString(),
            url = DataFactory.randomString(),
        )

    internal fun makeApiLocation(): ApiLocation =
        ApiLocation(
            name = DataFactory.randomString(),
            url = DataFactory.randomString(),
        )

    internal fun makeLocation(): ModelLocation =
        ModelLocation(
            name = DataFactory.randomString(),
            url = DataFactory.randomString(),
        )
}
